### ISSUE TITLE
Argo-Workflows: Helm chart do not seem to cleanly support "managed namespace" installation

### ISSUE THREAD

**Issue by hyacin75** [argoproj/argo-helm#1431]
### Describe the bug

The ["managed namespace" docs](https://argoproj.github.io/argo-workflows/managed-namespace/) say -

`You can install Argo in either cluster scoped or namespace scope configurations. **This dictates if you must set-up cluster roles or normal roles.**

In namespace scope configuration, you must run both the Workflow Controller and Argo Server using --namespaced. **If you would like to have the workflows running in a separate namespace**, add --managed-namespace **as well**. (In cluster scope installation, don't include --namespaced or --managed-namespace.)`

So that implies no cluster roles should be needed, or created in a managed namespace installation.

The chart values support no explicit definition of 'managed namespace' mode though, simply this -

`  # -- Specify all namespaces where this workflow controller instance will manage
  # workflows. This controls where the service account and RBAC resources will
  # be created. Only valid when singleNamespace is false.
  workflowNamespaces:
    - default
`

but, when singleNamespace is false, even with every other cluster related setting disabled, we still get CRs and CRBs -

`me@here:~/argo-helm/charts/argo-workflows$ helm template argo-workflows . | grep -i clusterrole | wc -l
15
me@here:~/argo-helm/charts/argo-workflows$ helm template argo-workflows . --set createAggregateRoles=false --set controller.clusterWorkflowTemplates.enabled=false --set server.clusterWorkflowTemplates.enabled=false --set server.clusterWorkflowTemplates.enableEditing=false | grep -i clusterrole | wc -l
6
me@here:~/argo-helm/charts/argo-workflows$ helm template argo-workflows . --set createAggregateRoles=false --set controller.clusterWorkflowTemplates.enabled=false --set server.clusterWorkflowTemplates.enabled=false --set server.clusterWorkflowTemplates.enableEditing=false | grep -i clusterrolebinding | wc -l
2
me@here:~/argo-helm/charts/argo-workflows$`

and setting singleNamespace to true would mean roles and rolebindings wouldn't be created for the additional namespace(s) listed in controller.workflowNamespaces as it explicitly says it is ignored if that is set.

### Related helm chart

argo-workflows

### Helm chart version

0.17.1

### To Reproduce

helm template argo-workflows . --set createAggregateRoles=false --set controller.clusterWorkflowTemplates.enabled=false --set server.clusterWorkflowTemplates.enabled=false --set server.clusterWorkflowTemplates.enableEditing=false

### Expected behavior

No CRs or CRBs since, as the docs say, they should only be created in cluster scoped installations.

**Comment by hyacin75** [comment]
In case something in unclear, as the 'confused' emote above leads me to believe -

If singleNamespace is false and workflowNamespaces has a value, logically you're looking for a "Managed Namespace" install, as, if you're doing a cluster install, there is no need or point to specifying any additional namespaces, as it gets access to ALL the namespaces via the CRs anyway.

In surely many organizations, any tool having access to ALL the namespaces is probably not ideal, but the "Managed Namespace" model is, where the workflows themselves, and the controller and server, can be separated.  As-is, the helm chart does not seem to support that.

**Comment by ju187** [comment]
For the current helm for role, etc if singleNamespace is set to true, tole, and sa are not only created in the managed namespaces, I assuming they should be

**Comment by vitalyrychkov** [comment]
The managed-namespace seems to be still not implemented in v3.4.7

**Comment by jcogilvie** [comment]
I agree this does not seem to work correctly.  There's no mention in the chart of the `--managed-namespace` flag indicated in that documentation blurb.

**Comment by pchar** [comment]
Hi someone has found a solution to this topic ?

**Comment by hyacin75** [comment]
> Hi someone has found a solution to this topic ?

I spent a couple days modifying the chart myself for my purposes.  It wasn't clean enough to submit a PR though sadly, just a quick and dirty 'get it working'.  It's likely horribly out of date now too and missing important stuff from recent updates.

**Comment by pchar** [comment]
Every help is welcome if you don't mind can you paste you quick and dirty ?  
it will give me some hits to fix ;)

**Comment by hyacin75** [comment]
> Every help is welcome if you don't mind can you paste you quick and dirty ? it will give me some hits to fix ;)

Copy/paste 100% no as it's got a lot of proprietary work stuff in it now.  When I get back to the office next week I can check my Confluence pages though to see if I have anything of use.  I know I was trying to work on a diff at one point so we could easily apply the changes to future chart versions ... I just don't think I ever finished :-/

**Comment by pchar** [comment]
Thanks ! any help is appreciated ;) 
have a nice we

**Comment by Stolz** [comment]
It's 2025. I would like to know if there is any clarification in this topic.
